src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/ErrorNotificationServiceTest.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.boot.test.rule.OutputCapture;
+import uk.gov.hmcts.reform.bulkscanprocessor.client.ErrorNotificationClient;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.in.ErrorNotificationResponse;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.out.errors.ErrorNotificationRequest;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.out.msg.ErrorCode;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.out.msg.ErrorMsg;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ErrorNotificationServiceTest {
+
+    @Rule
+    public OutputCapture capture = new OutputCapture();
+
+    @Captor
+    public ArgumentCaptor<ErrorNotificationRequest> requestCaptor;
+
+    @Mock
+    private ErrorNotificationClient client;
+
+    private ErrorNotificationService service;
+
+    @Before
+    public void setUp() {
+        capture.reset();
+        service = new ErrorNotificationService(client);
+    }
+
+    @After
+    public void tearDown() {
+        capture.flush();
+    }
+
+    @Test
+    public void should_send_notification_to_client_and_log_the_response() {
+        // given
+        ErrorMsg serviceBusMessage = new ErrorMsg(
+            "some id",
+            "zip file name",
+            "po box",
+            "document control number",
+            ErrorCode.ERR_AV_FAILED,
+            "antivirus flag",
+            true
+        );
+        ErrorNotificationResponse response = new ErrorNotificationResponse("notify id");
+        given(client.notify(requestCaptor.capture())).willReturn(response);
+
+        // when
+        service.processServiceBusMessage(serviceBusMessage);
+
+        // then
+        ErrorNotificationRequest request = requestCaptor.getValue();
+        assertThat(request).extracting(
+            "zipFileName", "poBox", "documentControlNumber", "errorCode", "errorDescription", "referenceId"
```

Comment:
why not just 
```
assertThat(request.zipFileName).isEqualTo(serviceBusMessage.zipFileName);
assertThat(request.poBox).isEqualTo(serviceBusMessage.poBox);
...
```
?
It's strongly typed and clearer.
You can use soft assertions to accumulate all assertion errors if you want.

Reply:
because request fields are private and final without any access to them. didn't change model just to accommodate unit test

Reply:
it's great that they're final, but why is there no way to read them??

Reply:
no need for them to be publicly accessed. request object is pretty much fire-and-forget principle. but can alter to public ;)